### ISSUE TITLE
Duration & Effort of Introduction to Programming courses problem

### ISSUE THREAD

**Issue by hematogender** [ossu/computer-science#613]
Hi there!

Course | Duration (Platform) |  Effort (Platform) | Duration (Repo) | Effort (Repo)
-------- | :---------------------: | :---------------: | :---------:  | :--------------------: 
[Python for Everyone](https://www.coursera.org/specializations/python) | 4 months | 12 hours/week | 9 weeks | 58 hours
[Fundamentals of Computing](https://www.coursera.org/specializations/computer-fundamentals) | 8 months | 6 hours/week | 9 weeks | 138 hours

I wonder, where do Duration and Effort numbers for both of Introduction to Programming courses come from?

For example, Coursera suggests 4 months with 12 hours/week effort for [Python for Everyone](https://www.coursera.org/specializations/python), while it is 9 weeks with 58 hours total effort<sup>1</sup> in the repository. That's ~6 hours/week.

4 months ≈ 17 weeks times 12 ≈ 200 hours total. Numbers are really really off here, or did I miss something?

Calculations for [Fundamentals of Computing](https://www.coursera.org/specializations/computer-fundamentals): 8 months ≈ 35 weeks times 6 ≈ 210 hours total. In the repo it is 9 weeks and 138 hours, which is ~15 hours/week.

<sup>1</sup> Why? Hours/week is used everywhere except in this category. I think, it is more convenient to use what a course platform suggests.

**Comment by waciumawanjohi** [comment]
The overall effort was calculated from adding up the hours listed on each course within the specialization*. So for example, the first course in Python for Everything states that it will take 12 hours to complete.

The inclusion of the duration was an oversight. Historically OSSU has listed a duration alongside hours/week. However, many students report working on classes full time. With an overall effort number and knowledge of the number of hours they can devote a week, students can better estimate how long it will take to complete a course. I'm going to remove the duration column from these two courses.

* I did not include the time for the 4 hour capstone test of Fundamentals of Computing.

**Comment by hematogender** [comment]
Sounds reasonable, thanks for a quick reply! I've done a quick check, it seems other courses are OK, except I found a similar problem in Core CS section: first two courses &mdash;How to Code&mdash; are 8-10 hours instead of 2-3 hours stated on their pages. Should we change that also?

**Comment by waciumawanjohi** [comment]
It sounds reasonable to me that we change How to Code to 2-3 hours a week, as that is what the course instructors advertise.

@hanjiexi I see that you estimated the courses at 8-10 hours a week in this commit:
https://github.com/ossu/computer-science/commit/9552eb9d4717334d76eb968e20a726b20f532e49?diff=split
Is there any background we should know before changing it?

**Comment by Isoaxe** [comment]
Actually I think the 2-3 hours for "How to Code" is a mistake. I'm currently doing this course, and have contacted edX about rectifying it.

The actual duration is per the syllabus section of the course, accessible after course commencement. This works out as 44 - 66 hours total over 6 weeks. So the currently stated duration is closer to the truth.

See below for details of the first part (simple data).

![Syllabus Course Duration](https://user-images.githubusercontent.com/20992024/67751397-a0eb6b00-fa63-11e9-8db5-1a07ba925298.png)

**Comment by hanjiexi** [comment]
@waciumawanjohi I think I generally went with the advertised time duration unless, after enrolling in the course, I found evidence to the contrary. What @Isoaxe found is the likely reason I went with a higher estimate. I also notice that I gave an estimated length of 60 hours on the [wiki page](https://github.com/ossu/computer-science/wiki/Core-Programming-draft) for Core Programming, excluding the "duration" column for the same reasons given above.

Realistically, if someone is already comfortable with programming, How to Code can be breezed through much quicker than 60 hours. I would consider it an estimated maximum rather than an average.
